force-app/main/default/classes/ChatterGroupLookupResultVMapper.cls

```diff
@@ -0,0 +1,116 @@
+ * @group
+ * @group-content ../../ApexDocContent/
+ * @description View Mapper class for Chatter Group Lookup Search functionality in EDA settings
+ */
+public virtual with sharing class ChatterGroupLookupResultVMapper {
+    /*********************************************
+     * @description Instance for Singleton Pattern
+     **********************************************/
+    @TestVisible
+    private static ChatterGroupLookupResultVMapper instance;
+
+    /*********************************************************************
+     * @description Empty constructor is not public for Singleton Pattern
+     **********************************************************************/
+    protected ChatterGroupLookupResultVMapper() {
+    }
+
+    /******************************************************************************
+     * @description Static method to get the current instance for Singleton pattern
+     * @return The instance of ChatterGroupLookupResultVMapper.
+     ******************************************************************************/
+    public static ChatterGroupLookupResultVMapper getInstance() {
+        if (instance == null) {
+            instance = new ChatterGroupLookupResultVMapper();
+        }
+
+        return instance;
+    }
+
+    /************************************************************************************************************
+     * @description Retrieve a list of sorted LookupResults representing a chatter group
+     * @param chatterGroupNameMatch The string to match chatter group names to
+     * @param recordLimit The number of records to limit the query to
+     * @return A list of Lookup Results with a name matching the search string
+     ************************************************************************************************************/
+    public virtual List<LookupResult> getLookupResultsLikeName(String chatterGroupNameMatch, Integer recordLimit) {
+        List<ChatterGroupModel> chatterGroupModelList = this.locateChatterGroupService()
+            .getChatterGroupModelsLikeName(chatterGroupNameMatch, recordLimit);
+        List<LookupResult> lookupResultList = new List<LookupResult>();
+
+        for (ChatterGroupModel ChatterGroupModel : chatterGroupModelList) {
+            lookupResultList.add(this.getLookupResultFromChatterGroupModel(ChatterGroupModel));
+        }
+
+        lookupResultList.sort();
+        return lookupResultList;
+    }
+
+    /************************************************************************************************************
+     * @description Creates a lookup result from a chatter group model
+     * @param ChatterGroupModel The chatter group model to convert to a Lookup Result
+     * @return The lookup result
+     ************************************************************************************************************/
+    public virtual LookupResult getLookupResultFromChatterGroupModel(ChatterGroupModel chatterGroupModel) {
```

Comment:
In this case we're performing a view model mapping. I wanted to separate the concerns and explicitly call out the nuances of this SObject mapping, particularly because it includes an icon and alternative text, which are business logic oriented and not quite appropriate to a view model consturctor.